### PR TITLE
Use activeFedora current solr connection for the extract uri

### PR DESCRIPTION
**Pull request by jcoyne**
Previously we were using the value of the yaml configuration, which is
the default way to initialize a connection, but not the only way. It is
also possible for the connection to be initialized programatically.
This is what hyku is doing to support multiple tenants each with their
own solr collection.

See https://github.com/samvera-labs/hyku/issues/1275

### CODE REVIEW

File: lib/hydra/derivatives/processors/full_text.rb [hunk 1]
```diff
@@ -50,11 +50,12 @@ def original_size
 
       # @returns [URI] path to the extract service
       def uri
-        @uri ||= URI("#{connection_url}/update/extract?extractOnly=true&wt=json&extractFormat=text")
+        @uri ||= connection_url + 'update/extract?extractOnly=true&wt=json&extractFormat=text'
```
Comment (mjgiarlo) [inline lib/hydra/derivatives/processors/full_text.rb]: Why the change from string interpolation to appending? Don't have a strong objection -- mostly curious.

Comment (jcoyne) [inline lib/hydra/derivatives/processors/full_text.rb]: @mjgiarlo previously `connection_url` returned a string. Now it returns a URI object.

**Comment by dlpierce** [comment]
I made a similar change as part of another issue, but hadn't gotten back around to PR it. Looks good.
